fiftyone/core/models.py

```diff
@@ -507,14 +508,15 @@ def save_batch(sample_batch, labels_batch):
                 if isinstance(imgs, Exception):
                     raise imgs
 
+                image_sizes = imgs.pop("fo_image_size", [(None, None)])
                 if needs_samples:
-                    labels_batch = model.predict_all(
+                    output = model.preprocess_and_forward_pass(
```

Comment:
As @jacobsela pointed out, this is a sensitive change. The `Model` interface is a public-facing thing that was created so that users can write their own `Model` instances that implement our interface so that `apply_model()` can support them:
https://docs.voxel51.com/model_zoo/design.html